### PR TITLE
feat!: Migrate firebase_dynamic_links to sound null safety

### PR DESCRIPTION
**Pull request by AlexHartford**
## Description

This PR updates firebase dynamic links to sound null safety.

The primary uncertainty is integration_test package consuming e2e. It appears flutter_driver is holding packages back. It also appeared in other null-safe releases of other flutterfire packages (e.g., firebase_core), the integration tests were removed (not sure if this is a temporary measure or not). Hence, I followed suit and removed the dependency on e2e & flutter_driver.

## Related Issues

https://github.com/FirebaseExtended/flutterfire/discussions/4182

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`).
This will ensure a smooth and quick review process. Updating the `pubspec.yaml` and changelogs is not required.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] My PR includes unit or integration tests for *all* changed/updated/fixed behaviors (See [Contributor Guide]).
- [x] All existing and new tests are passing.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] The analyzer (`flutter analyze`) does not report any problems on my PR.
- [x] I read and followed the [Flutter Style Guide].
- [x] I signed the [CLA].
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require plugin users to manually update their apps to accommodate your change?

- [x] Yes, this is a breaking change (please indicate a breaking change in CHANGELOG.md and increment major revision).
- [ ] No, this is *not* a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/FirebaseExtended/flutterfire/blob/master/CONTRIBUTING.md
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning
[CLA]: https://cla.developers.google.com/

### CODE REVIEW

**Comment by Salakar** [comment]
cc @rrousselGit - would you mind reviewing this one, thanks!

File: packages/firebase_dynamic_links/CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,7 @@
+## 1.0.0
+
+- **BREAKING** **FEAT**: Upgrade to sound null-safety.
+
```
Comment (rrousselGit) [inline packages/firebase_dynamic_links/CHANGELOG.md:4]:
```suggestion
```

Comment (rrousselGit) [reply packages/firebase_dynamic_links/CHANGELOG.md]: (for reference, changelogs are automatically generated)

**Comment by rrousselGit** [comment]
@googlebot I consent.

**Comment by rrousselGit** [comment]
The code looks good, but I'm confused as to why the CI doesn't start properly.

**Review by rrousselGit — APPROVED** [review]
It appears that this is expected. LGTM then!

**Pull request merged**